fission-web-server/library/Fission/Web/Server/Types.hs

```diff
@@ -140,6 +155,102 @@ instance MonadDB (Transaction Server) Server where
 instance MonadRelayStore Server where
   getStoreVar = asks linkRelayStoreVar
 
+instance MonadPowerDNS Server where
+  getClientEnv = do
+    apiUrl <- asks pdnsURL
+    apiKey <- asks pdnsApiKey
+    uri <- parseBaseUrl $ Text.unpack (textDisplay apiUrl)
+    let
+      rawHTTPSettings =
+        case baseUrlScheme uri of
+          Http  -> defaultManagerSettings
+          Https -> tlsManagerSettings
+
+    mgr <- liftIO $ newManager rawHTTPSettings
+    return $ PDNS.applyXApiKey (textDisplay apiKey) (mkClientEnv mgr uri)
```

Comment:
We should move most of this to `Main.hs`, and thread the HTTP manager through in the `Config`. This is setting up a connection pool, just like with Postgres. So right now, every time you make a DNS call, we're doing all that work over again.

Having a `getClientEnv` method is totally fine, if you want that to be part of the public API, but change it to be something like `asks pdnsManager`. I would probably remove it from the public API, personally, and create a helper function in this module that gets called in the class methods.

Reply:
I think I've done this now.